### PR TITLE
Implement MetricProducer directly by the SdkMeterProvider

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java [hunk 1]
```diff
@@ -58,20 +61,14 @@ public SdkMeter get(String instrumentationName, @Nullable String instrumentation
...
-   *
-   * <p>WARNING: A MetricProducer is stateful. It will only return changes since the last time it
-   * was accessed. This means that if more than one {@link
-   * io.opentelemetry.sdk.metrics.export.MetricExporter} has a handle to this MetricProducer, the
-   * two exporters will not receive copies of the same metric data to export.
-   *
-   * @return the {@link MetricProducer} that can be used to retrieve metrics from this {@code
-   *     MeterSdkProvider}.
-   */
-  public MetricProducer getMetricProducer() {
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java:73]: I know metrics is alpha still, but is it worth deprecating this method (and changing it to return `this`) for one release, to ease the transition for alpha metrics users?

Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java:73]: Your call ... I can do that easily do let's do it

Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java:73]: Done.

**Review by jkwatson — APPROVED** [review]
👍

**Pull request merged**